fix: realign tax rate inputs across region and tax settings

kasperkristensen opened the pull request:
**What**
- Ensure a uniform experience when adding tax rates in both region and tax settings.

Closes CORE-559

olivermrbl reviewed (changes requested):
A couple of comments:
- I can't seem to add tax rates with decimals
- Tax rate and tax code are not editable in "Edit Region Details" modal
- Tax rate and tax code should be renamed to "Default Tax Rate" and "Default Tax Code" in regions setting - both create and update

kasperkristensen commented:
> A couple of comments:
> 
> * I can't seem to add tax rates with decimals
> * Tax rate and tax code are not editable in "Edit Region Details" modal
> * Tax rate and tax code should be renamed to "Default Tax Rate" and "Default Tax Code" in regions setting - both create and update

Addressed feedback, with the exception of `Tax rate and tax code are not editable in "Edit Region Details" modal`, as per Seb this should only be edited from the tax settings.

olivermrbl reviewed (approved):
This is great! 💪 

Noticed a minor thing. Countries in "Create Region" flow should be required. If left out, page breaks when trying to map through undefined countries.

Feel free to merge afterwards!
